components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts

```diff
@@ -8,7 +8,8 @@ import { NoScriptInfo } from '../other/noScriptInfo'
 import { SettingsData } from '../other/settingsTypes'
 
 export interface Tab {
-  cosmeticBlocking: boolean
+  cosmeticFiltering: BlockFPOptions
+  cosmeticFilteringFlag: boolean
```

Comment:
why do we need a flag if we already have the setting?

Reply:
I believe the idea is to add a way to decouple cosmetic filtering from network filtering, even though for general use cases they'll be switched together. cc @pes10k

Reply:
Hmm, im not sure I follow the above. We do want to be able to control cosmetic filtering independent of network filtering, but I don't follow why there are two different `cosmeticFiltering` controls above

Reply:
@pes10k `cosmeticFilteringFlag` tracks the state of the feature flag. `cosmeticFiltering` tracks whether cosmetic filtering is for third-party only, everything, or completely off when the flag is on. Will rename/document to be more clear after discussion with @bridiver.